### ISSUE TITLE
[Feature Request] "Tag search" as the first point of entry

### ISSUE THREAD

**Issue by orionx** [Razeeman/Android-SimpleTimeTracker#262]
Hi Razeeman,

First of all, thank you for creating this fantastic app. I’ve been a loyal user for over a year now, and it has been incredibly helpful in organizing my work and personal tasks.

Over time, I’ve accumulated hundreds of tags across multiple categories. Some categories have over a hundred tags, and certain tags are shared across categories while others are not. At this scale, searching through the tags to find the right one has become quite time-consuming and challenging, especially when tags span across multiple categories.

It would be amazing if you could implement a **"tag search" feature as the first point of entry**. The search result list would display all the categories and tags that match the search keyword, allowing users to quickly select the correct Category+Tag, enabling faster and more efficient input of new entries.

I truly believe this enhancement would improve the user experience, especially for those of us managing a large number of tags.

Thank you again for your hard work and for considering this suggestion. I’d be happy to provide any additional input, if needed.

**Comment by Razeeman** [comment]
Hello! Glad you liked the app. Can you explain a bit more what is meant by the "first point of entry"?

**Comment by Cancoillotte** [comment]
I agree, it's a fantastic application. As far as I'm concerned, it's my life companion, I could say!

Reading orionx's post, I wonder if I wouldn't use comments the way others use tags. Because I have very few tags, but many more comments, which are often reused.

Would it be possible to link the search for comments to orionx's request? Like the possibility of finding the periods with this or that comment?

Could the “first point of entry” be one more gray button, next to the others “Repeat, Add, Default”?

**Comment by Razeeman** [comment]
@Cancoillotte I assumed the issue is that when there are many tags created, it is hard to find right one when starting activity. So I was planning to add a search for tags to the tag selection dialog and on the record edit screen. For comments same search is already available on the records edit screen and will be added to the tag / comment selection dialog in #261

Is this what you are looking for? 

By the way, finding records with some comment is available from the filter on the detailed statistics screen.

**Comment by Cancoillotte** [comment]
Yes, great, that's exactly it. I admit I haven't mastered all the application's features yet, there are so many possibilities!

But I don't understand #261 at all: the discussion dates from December 16, and I have the impression that this feature is already present in version 1.47 of November 27...

As for the search for comments in the filters, I hadn't noticed it yet. It's great that it's already there, and that's what I was looking for, great! But I think it's a shame that there's a limit: you have to select an activity or category first. I'd like to be able to access them directly, as with activities or categories. But this detail isn't extremely important, for me at least, it's just a suggestion for improvement.

I hadn't noticed the search for comments in the registration dialog either! But there's a problem, it doesn't work very well for me: if I look at the list of old comments, I only see them once each, which is normal. But when I do a search, as soon as I've typed just one letter, each comment containing that letter is displayed several times, I imagine as many times as it's already been used. This could become a problem as the number of times comments are used increases. But as far as I'm concerned, it doesn't bother me at the moment.

There is, however, a difference in operation that confused me when I tried it out: when you select an old comment directly, it appears in the input field at the same time as on the colored card at the top of the screen. But when you're in a search, it's not displayed in the search field, but only on the colored card. I didn't notice this at first, and thought it wasn't working.

In fact, I don't really understand the difference between tags and comments, in terms of use, even though it's certainly a good thing that there are both possibilities. And I wonder whether I'm using both features properly. For example, I wonder if I could use tags instead of comments, or vice versa. What's the original idea?

I hope I'm not too confused in my explanations.

**Comment by Razeeman** [comment]
Added tag search when editing a record or selecting a tag on timer start. Will be in the next release. Hope this helps with this issue.

**Comment by Razeeman** [comment]
> I wonder if I could use tags instead of comments, or vice versa. What's the original idea?

Probably comments can be used instead of comments, but tags have more statistics.

**Comment by orionx** [comment]
> > I wonder if I could use tags instead of comments, or vice versa. What's the original idea?
> 
> Probably comments can be used instead of comments, but tags have more statistics.

Yes, that's exactly why I requested this feature. Tags offer more structure and useful statistics than comments. Thank you so much for making it happen, Razeeman! I really appreciate it.
